### ISSUE TITLE
Unable to start instances with DATA volumes of deploy-as-is template having instance snapshots 

### ISSUE THREAD

**Issue by rajujith** [apache/cloudstack#9180]
##### ISSUE TYPE

 * Bug Report
 

##### COMPONENT NAME

~~~
Template, VMware
~~~

##### CLOUDSTACK VERSION

~~~
4.19
~~~

##### OS / ENVIRONMENT

Instance deployed from deplo-as-is template. VMware vSphere 8.0 

##### SUMMARY

While starting an instance created from deploy-as-is template cloudstack tries to tear down the data disks. Since the instance has data volumes attached before the instance snapshot vCenter declines the teardown. This design issue results in failure to start an instance having data volumes and instance snapshots. 

```
2024-06-05 12:16:40,023 INFO  [c.c.h.v.r.VmwareResource] (DirectAgent-101:ctx-79cfdac2 10.1.32.212, job-38/job-40, cmd: StartCommand) (logid:8e044e50) Tearing down datadisks for deploy-as-is VM
2024-06-05 12:16:40,121 ERROR [c.c.h.v.r.VmwareResource] (DirectAgent-101:ctx-79cfdac2 10.1.32.212, job-38/job-40, cmd: StartCommand) (logid:8e044e50) Error mapping deploy-as-is VM disks from cloned VM i-2-3-VM
java.lang.RuntimeException: Invalid configuration for device '0'.
        at com.cloud.hypervisor.vmware.util.VmwareClient.waitForTask(VmwareClient.java:426)
```

##### STEPS TO REPRODUCE

~~~
1. Deploy an instance from a deploy-as-is template with a data volume( Read Instance settings from OVA selected for template).
2. Create an instance snapshot
3. Stop and then start the instance.
~~~

##### EXPECTED RESULTS

~~~
The user should be able to perform stop-and-start operations on instances deployed from deploy-as-is template types having data disks and instance snapshots. 
~~~

##### ACTUAL RESULTS

~~~
The user is unable to stop and start the instance. 
~~~

**Comment by weizhouapache** [comment]
I am able to reproduce the issue by the following steps

 - register a deploy-as-is template
 - create a vm
 - create a datadisk and attach to the vm
 - create vm snapshot
 - stop the vm
 - start the vm, and got the same error

note
- on vcenter, it shows " Invalid configuration for device '0'. Cannot remove virtual disk from the virtual machine because it or one of its parent disks is part of a snapshot of the virtual machine."
- vm can be started successfully in vcenter
- vm can be started successfully in cloudstack, after removing the vm snapshot.

it looks the issue is caused by

https://github.com/apache/cloudstack/blob/66f7f76263619c420f922444a1679d2f992c26ab/plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java#L2835-L2838

which is called by

https://github.com/apache/cloudstack/blob/66f7f76263619c420f922444a1679d2f992c26ab/plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/vmware/resource/VmwareResource.java#L2204-L2207

the code was introduced in #4307 , do you remember why added it and have any idea how to improve it ? cc @nvazquez

**Comment by nvazquez** [comment]
Hi @weizhouapache as I do recall there was a similar logic for non-deploy as is VMs but reviewing the code I see those are checking for snapshots. I will review your linked PR

**Comment by DaanHoogland** [comment]
fixed in #9243
